From 7b029f41cacefdfc79d79d78afed631d9191a799 Mon Sep 17 00:00:00 2001
From: Alexandru M Stan <alex@hypertriangle.com>
Date: Thu, 8 Oct 2020 23:55:26 -0500
Subject: [PATCH 25/68] power: supply: some: Fix battery math

This gives us some more realistic numbers from the battery

Signed-off-by: Steev Klimaszewski <steev@kali.org>
Co-Developed-by: Steev Klimaszewski <steev@kali.org>
---
 drivers/power/supply/some-battery.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/power/supply/some-battery.c b/drivers/power/supply/some-battery.c
index 43a8f6e0667c8..44ea5a27e7335 100644
--- a/drivers/power/supply/some-battery.c
+++ b/drivers/power/supply/some-battery.c
@@ -193,7 +193,7 @@ static int some_battery_update_status(struct some_battery *battery)
 
 	lsb = some_battery_read(battery, BAC0);
 	msb = some_battery_read(battery, BAC1);
-	battery->rate_now = sign_extend32((msb << 8) | lsb, 16);
+	battery->rate_now = sign_extend32((msb << 8) | lsb, 15) * battery->basc;
 
 	if (battery->unit_ma)
 		battery->rate_now = battery->rate_now * battery->voltage_now / 1000;
@@ -279,16 +279,16 @@ static int bat0_get_property(struct power_supply *psy,
 		val->intval = battery->design_voltage;
 		break;
 	case POWER_SUPPLY_PROP_CHARGE_FULL_DESIGN:
-		val->intval = battery->design_capacity;
+		val->intval = battery->design_capacity * 100;
 		break;
 	case POWER_SUPPLY_PROP_CHARGE_FULL:
-		val->intval = battery->full_charge_capacity;
+		val->intval = battery->full_charge_capacity * 100;
 		break;
 	case POWER_SUPPLY_PROP_CHARGE_NOW:
-		val->intval = battery->capacity_now;
+		val->intval = battery->capacity_now * 100;
 		break;
 	case POWER_SUPPLY_PROP_CURRENT_NOW:
-		val->intval = battery->rate_now;
+		val->intval = battery->rate_now * 100;
 		break;
 	case POWER_SUPPLY_PROP_VOLTAGE_NOW:
 		val->intval = battery->voltage_now;
-- 
2.26.2

